pkg/antctl/raw/supportbundle/command.go

```diff
@@ -692,9 +702,151 @@ func processResults(resultMap map[string]error, dir string) error {
 		err = writeFailedNodes(dir, failedNodes)
 	}
 
+	// download logs from kubernetes api
+	if failedNodes != nil {
+		err = downloadAgentBundleFromKubernetes(ctx, antreaClientset, k8sClient, failedNodes, dir)
+		if err != nil {
+			fmt.Println("Failed to download agent bundle from kubernetes api: " + err.Error())
+		} else {
+			allFailed = false
+		}
+	}
+	if controllerFail {
+		err = downloadControlleBundleFromKubernetes(ctx, antreaClientset, k8sClient, dir)
+		if err != nil {
+			fmt.Println("Failed to download controller bundle from kubernetes api: " + err.Error())
+		} else {
+			allFailed = false
+		}
+	}
+
 	if allFailed {
 		return fmt.Errorf("no data was collected: %s", genErrorMsg(resultMap))
 	} else {
 		return err
 	}
 }
+
+func downloadControlleBundleFromKubernetes(ctx context.Context, antreaClientset antrea.Interface, k8sClient kubernetes.Interface, dir string) error {
```

Comment:
typo in the function name ("Controlle" instead of "Controller")

I would also recommend something more explicit: `downloadFallbackControllerBundleFromKubernetes` and `downloadFallbackAgentBundleFromKubernetes`